13-multiplecomponents-lab/src/02-shoppinglist.jsx

```diff
@@ -18,11 +18,183 @@
  *
  */
 
-function ShoppingItemsCount(props) {
-  return <p>Total items: {props.count}</p>
+import React from 'react';
+
+export default class ShoppingCart extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            items: [],
+            currentItem: {}
+        }
+    }
+
+    addItem = () => {
+        let items = this.state.items;
+        let newItem = {};
+        items.push(this.state.currentItem);
```

Comment:
you can use ES6 array spread for this:
The following will be equivalent of creating new array and pushing new item to it. Also destructure items and currentItem from state.

```
addItem = () => {
    let { items, currentItem } = this.state;
    this.setState({
      items: [ ...items, currentItem ],
      currentItem: {}
    });
  }
```